### PR TITLE
Add depfile to configure_file()

### PR DESCRIPTION
**Pull request by elmarco**
In qemu, minikconf generates a depfile that meson could use to
automatically reconfigure on dependency change.

Note: someone clever can perhaps find a way to express this with a ninja rule & depfile=. I didn't manage, so I wrote a simple depfile parser.

### CODE REVIEW

**Comment by bonzini** [comment]
Ping (@elmarco can you refresh to fix the CI failure?)

**Comment by elmarco** [comment]
ping (@xclaesse ?)

File: mesonbuild/interpreter.py [hunk 1]
```diff
@@ -3742,6 +3754,9 @@ def func_configure_file(self, node, args, kwargs):
             # that the command is run from is 'unspecified', so it could change.
             # Currently it's builddir/subdir for in_builddir else srcdir/subdir.
             values = mesonlib.get_filenames_templates_dict(inputs_abs, [ofile_abs])
+            if depfile:
+                depfile = os.path.join(self.environment.build_dir, depfile)
```
Comment (jpakkane) [inline mesonbuild/interpreter.py]: This should be `self.environment.get_scratch_dir()` rather than `self.environment.build_dir`.

**Comment by jpakkane** [comment]
Featurewise this is ok.

File: mesonbuild/interpreter.py (hunk 1 above)
Comment (elmarco) [reply mesonbuild/interpreter.py]: make sense, done

**Pull request merged**